src/flows/neuralspline.jl

```diff
@@ -0,0 +1,235 @@
+    px, py, dydx = get_nsc_params(nsc, y2)
+    y1 = _ensure_matrix(y1)
+    x1, logjac = MonotonicSplines.rqs_inverse(y1, px, py, dydx)
+    return Bijectors.combine(nsc.mask, vec(x1), y2, y3), logjac[1]
+end
+function Bijectors.with_logabsdet_jacobian(insl::Inverse{<:NeuralSplineCoupling}, y::AbstractMatrix)
+    nsc = insl.orig
+    y1, y2, y3 = partition(nsc.mask, y)
+    px, py, dydx = get_nsc_params(nsc, y2)
+    x1, logjac = MonotonicSplines.rqs_inverse(y1, px, py, dydx)
+    return Bijectors.combine(nsc.mask, x1, y2, y3), vec(logjac)
+end
+
+function (nsc::NeuralSplineCoupling)(x::AbstractVecOrMat)
+    return Bijectors.transform(nsc, x)
+end
+
+
+"""
+    NSF_layer(dims, hdims, K, B; paramtype = Float64)
+
+Build a single Neural Spline Flow (NSF) layer by composing two
+`NeuralSplineCoupling` bijectors with complementary odd–even masks.
+
+Arguments
+- `dims::Int`: dimensionality of the problem.
+- `hdims::AbstractVector{Int}`: hidden sizes of the conditioner network.
+- `K::Int`: number of spline knots.
+- `B::AbstractFloat`: spline boundary.
+
+Keyword Arguments
+- `paramtype::Type{T} = Float64`: parameter element type.
+
+Returns
+- A `Bijectors.Bijector` representing the NSF layer.
+
+Example
+- `layer = NSF_layer(4, [64,64], 10, 3.0)`
+- `y = layer(randn(4, 32))`
+"""
+function NSF_layer(
+    dims::T1,                      # dimension of problem
+    hdims::AbstractVector{T1},     # dimension of hidden units for nn 
+    K::T1,                           # number of knots
+    B::T2;                           # bound of the knots
+    paramtype::Type{T2} = Float64,   # type of the parameters
+) where {T1<:Int,T2<:AbstractFloat}
+
+    mask_idx1 = 1:2:dims
+    mask_idx2 = 2:2:dims
+
+    # by default use the odd-even masking strategy
+    nsf1 = NeuralSplineCoupling(dims, hdims, K, B, mask_idx1, paramtype)
+    nsf2 = NeuralSplineCoupling(dims, hdims, K, B, mask_idx2, paramtype)
+    return reduce(∘, (nsf1, nsf2))
+end
+
+"""
+    nsf(q0, hdims, K, B, nlayers; paramtype = Float64)
+    nsf(q0; paramtype = Float64)
+
+Construct an NSF by stacking `nlayers` `NSF_layer` blocks. The one-argument
+variant defaults to 10 layers with `[32, 32]` hidden sizes, 10 knots, and
+boundary `30` (scaled by `one(T)`).
+
+Arguments
+- `q0::Distribution{Multivariate,Continuous}`: base distribution.
+- `hdims::AbstractVector{Int}`: hidden sizes of the conditioner network.
+- `K::Int`: spline knots per coordinate.
+- `B::AbstractFloat`: spline boundary.
+- `nlayers::Int`: number of NSF layers.
+
+Keyword Arguments
+- `paramtype::Type{T} = Float64`: parameter element type.
+
+Returns
+- `Bijectors.TransformedDistribution` representing the NSF flow.
+
+Notes:
+- Under the hood, `nsf` relies on the rational quadratic spline function implememented in 
+`MonotonicSplines.jl` for performance reasons.  `MonotonicSplines.jl` uses 
+`KernelAbstractions.jl` to support batched operations. 
+Because of this, so far `nsf` only supports `Zygote` as the AD type.
```

Comment:
There is a nice feature in `Documenter` that highlights important information: See the `!!!` series [here](https://documenter.juliadocs.org/stable/showcase/#Admonitions).